agent/tool-scripts/postprocess/mpstat-postprocess

```diff
@@ -59,11 +70,67 @@ sub calc_timestamp {
 	return $date_ms + $this_time;
 }
 
-# read the mpstat.txt and collect stats for all non dm-* devices
-open(MPSTAT_TXT, "$dir/mpstat-stdout.txt") || die "could not find $dir/mpstat-stdout.txt\n";
+sub cpu_is_online {
+	my $cpu = shift;
+	if (exists $online_cpus{$cpu}) {
+		return $online_cpus{$cpu};
+	} elsif (-e $dir . "/online-cpus.txt" and open(FH, $dir . "/online-cpus.txt")) {
+		# Try a saved file of online status for this mpstat (this
+		# relies on this file being generated at some point before
+		# postprocessing, likely at stop-tools phase),
+		# This may not exist depending on what version of pbench
+		# was used to get the data.
+		while (my $line = <FH>) {
+			# Take the opporunity to read all the CPUs and populate
+			# our hash
+			if ($line =~ /^(\d+):\s*([0-1])$/) {
+				$online_cpus{$1} = $2;
+			}
+		}
+		close(FH);
+		return $online_cpus{$cpu};
+	} elsif (not defined $hostname or $hostname eq chomp(my $this_host = `hostname -s`)) {
```

Comment:
It appears this logic means, "If this script was not invoked for CDM data processing, or if it was and the hostname given as an argument matches this hostname, then use the local system's `/sys/devices/system/cpu` tree to determine if CPUs are online."

It seems that this is going to be problematic, as re-processing the mpstat data will fall into this path when not using CDM.

Can we change this logic such that the caller of mpstat-postprocess can tell it the behavior to take, since it knows the context?

If this script is told that it is dealing with live data, fetch the online CPUs directly, else fetch the saved online CPUs file, failing if it doesn't have one.

Reply:
I think the part missing is not returning "1" when the online cpu file is missing, as we did in the original code.  And so when not returning "1", the CPU is assumed to not be online, and the data is not generated.  Just adding that back should resolve everything.  Adding more logic to the caller [indicating live or not live] is a possibility, but I think we're then adding a much more complex feature, which would likely need to be implemented for all tools, that I am not ready to do just yet.

Reply:
Let's fix that missing file behavior and see how this plays out.

The problem is that the unit tests will get screwed up in we run them on a system where `/sys` CPU file is available and it is offline.

I can take a crack at this next week if you don't have the time.

Reply:
That's a good point about a different /sys.  That would be great if you could work on it next week, as I will not here all next week.  I believe the commit I just pushed will fix the unittest, and I'll push another commit for the typo.

Reply:
I'll post updates in a PR that follows this one.

Reply:
Posted PR #1212 to address this situation.  Now mpstat-postprocess expects to find an `online-cpus.txt` file with the proper CPU IDs marked on- or off-line, and does not read from `/sys/devices/system/cpu` tree at all.